src/zenml/integrations/neptune/experiment_trackers/run_state.py

```diff
@@ -0,0 +1,105 @@
+import functools
+import hashlib
+
+import neptune.new as neptune
+
+from zenml.client import Client
+from zenml.integrations.constants import NEPTUNE
+
+
+def singleton(class_):
+    instances = {}
+
+    def getinstance(*args, **kwargs):
+        if class_ not in instances:
+            instances[class_] = class_(*args, **kwargs)
+        return instances[class_]
+
+    return getinstance
+
+
+class NoActiveRunException(Exception):
+    pass
+
+
+class InvalidExperimentTrackerSelected(Exception):
+    pass
+
+
+@singleton
+class RunState:
+    def __init__(self):
+        self._active_run = None
+        self._project = None
+        self._run_name = None
+        self._token = None
+        self._tags = None
+
+    @property
+    def project(self):
+        return self._project
+
+    @property
+    def token(self):
+        return self._token
+
+    @property
+    def run_name(self):
+        return self._run_name
+
+    @property
+    def tags(self):
+        return self._tags
+
+    @project.setter
+    def project(self, project):
+        self._project = project
+
+    @token.setter
+    def token(self, token):
+        self._token = token
+
+    @run_name.setter
+    def run_name(self, run_name):
+        self._run_name = run_name
+
+    @tags.setter
+    def tags(self, tags):
+        self._tags = tags
+
+    def set_active_run(self, run: neptune.metadata_containers.Run):
+        self._active_run = run
```

Comment:
This is not really a user-exposed interface so let's remove this function and made it just a statement in `active_run`